kong/db/strategies/cassandra/init.lua

```diff
@@ -60,6 +60,128 @@ local function extract_pk_values(schema, entity)
 end
 
 
+local function is_partitioned(self)
+  local cql
+
+  if self.connector.major_version == 3 then
+    cql = fmt([[
+      SELECT * FROM system_schema.columns
+      WHERE keyspace_name = '%s'
+      AND table_name = '%s'
+      AND column_name = 'partition'
+      AND kind = 'partition_key'
+      ALLOW FILTERING;
+    ]], self.connector.keyspace, self.schema.name)
+
+  else
+    cql = fmt([[
+      SELECT * FROM system.schema_columns
+      WHERE keyspace_name = '%s'
+      AND columnfamily_name = '%s'
+      AND column_name = 'partition'
+      ALLOW FILTERING;
+    ]], self.connector.keyspace, self.schema.name)
+  end
+
+  local rows = assert(self.connector:query(cql, {}, nil, "read"))
```

Comment:
our constructors always follow the `return nil, err` pattern, and never throw errors (except programmer errors). This should be returned all the way up to https://github.com/Kong/kong/blob/master/kong/db/strategies/init.lua#L40-L43

Reply:
Understood. Removed the `assert` and added a bunch of `if err then return nil, err end`